leaderElection/src/Node.java

```diff
@@ -99,8 +106,49 @@ private void processMsg(Msg msg) {
                 } else {
                     addMsgToBuffer(msg);
                 }
-            } else if(msg.getAction().equals(MsgAction.TEST)) {
+            } else if (msg.getAction().equals(MsgAction.TEST)) {
                 Logger.Debug("%s", msg.toString());
+            } else if (msg.getAction().equals(MsgAction.SENDSEARCH)) {
+                String content = msg.getContent();
+                int fromId = msg.getFromId();
+                if (content.equals("Search") || content.equals("Null")){
+                    if (!marked && msg.getRound() != getRound()) {
+                        addMsgToTreeBuffer(msg);
+                    } else if (msg.getRound() == getRound()){
+                        if (content.equals("Search")) {
+                            updateProcessedMsgNo();
+                            if (marked == false) {
+                                marked = true;
+                                setDegree(getRound());
+                                maxDegree = Math.max(degree, maxDegree);
+                                setParent(fromId);
+                                replySearchMsg("Accept", fromId);
+                            } else{
+                                replySearchMsg("Reject", fromId);
```

Comment:
I think is also necessary to create another two Msg.Action for "Accept" and "Reject".